### PR TITLE
allow Kafka Channel to use ns dispatcher based on annotation

### PR DESCRIPTION
**Pull request by lionelvillard**
Fixes #

## Proposed Changes

- Merge the kafka and kafka/namespace configs into one
- When `eventing.knative.dev/scope: namespace` annotation is set on the kafka channel resource, the underlying service is configured to forward events to the dispatcher deployed in the same namespace as the channel.
-

**Release Note**

```release-note
Kafka channels with the `eventing.knative.dev/scope: namespace` annotation uses the Kafka dispatcher deployed in the same namespace as the channel definition.
```

**Docs**

https://github.com/knative/docs/issues/2242

### CODE REVIEW

**Comment by lionelvillard** [comment]
Waiting for [this](https://github.com/knative/eventing/pull/2628) to land to use the const annotation key

File: kafka/channel/README.md [hunk 1]
```diff
@@ -125,14 +87,46 @@ kubectl get deployment -n knative-eventing kafka-webhook
...
 ```shell
 kubectl get configmap -n knative-eventing config-kafka
 ```
 
-- **namespace-scoped**:
+### Namespace Dispatchers
 
-```shell
-kubectl get configmap   config-kafka
+By default events are received and dispatched by a single cluster-scoped dispatcher components. You can also specify whether events should be received and dispatched by the dispatcher in the same namespace as the channel definition by adding the
```
Comment (matzew) [inline kafka/channel/README.md]: can I mix ? 

have one cluster wide?

and a few ns specific channels ? 

(sure, all backed by the same kafka broker)

File: kafka/channel/README.md [hunk 2]
```diff
@@ -125,14 +87,46 @@ kubectl get deployment -n knative-eventing kafka-webhook
...
+apiVersion: messaging.knative.dev/v1alpha1
+kind: KafkaChannel
+metadata:
+  name: my-kafka-channel
+  namespace: <YOUR_NAMESPACE>
+  annotations:
+    eventing.knative.dev/scope: namespace
+spec:
+  numPartitions: 1
+  replicationFactor: 3
```
Comment (matzew) [inline kafka/channel/README.md]: This requires 3 NOTES

our doc does only install small cluster w/ one node 


the same config error is on the snippet for the cluster wide disptacher 

@lionelvillard can you chagen ? 

`numPartitions`  can be larger than the number of nodes, but not the `replication`

File: kafka/channel/README.md (hunk 1 above)
Comment (lionelvillard) [reply kafka/channel/README.md]: yes you can mix and have a different Kafka Broker per NS

**Comment by matzew** [comment]
tested, and works 

```
default-broker-filter-748d697f58-sg48c   1/1     Running   0          3h58m
default-broker-ingress-bc748f8d7-794tf   1/1     Running   0          3h58m
kafka-ch-dispatcher-58ccc54d56-97pvx     1/1     Running   0          20s
```

and got a dispatcher in the cluster scoped `knative-eventig`

**Comment by matzew** [comment]
and I have these topics:

```
knative-messaging-kafka.blahtest.my-kafka-channel
knative-messaging-kafka.default.my-kafka-channel
```

**Pull request merged**